internal/fs/fs.go

```diff
@@ -2651,7 +2662,7 @@ func (fs *fileSystem) WriteFile(
 	}
 
 	// Serve the request.
-	err = in.Write(ctx, op.Data, op.Offset)
+	err = in.Write(ctx, op.Data, op.Offset, util.Write)
```

Comment:
line no: 2659 in a new change and not incorporated in the new flow. We need to fix this. Can you create a task for this, so we dont forget.